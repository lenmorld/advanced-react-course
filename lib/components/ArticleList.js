import React from 'react';

import Article from './article';

const ArticleList = (props) => {

  // for testing, have to know dependencies
  // e.g. props.articles

  return(
    <div>
      {Object.values(props.articles).map(article =>
        <Article
          key={article.id}
          article={article}
          actions={props.articleActions}
        />
      )}
    </div>
  );
};

export default ArticleList;
