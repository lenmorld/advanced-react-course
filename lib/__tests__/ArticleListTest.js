import React from 'react';
import ArticleList from '../components/ArticleList';

import renderer from 'react-test-renderer';
// yarn add --dev react-test-renderer
// yarn add --dev jest

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' },
    },
    // mock function
    articleActions: {
      lookupAuthor: jest.fn(() => ({})),
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <ArticleList
        {...testProps}
      />
    ).toJSON();

    console.log(tree);

    expect(tree.children.length).toBe(2);

    // snapshot expetation, tree must match snapshot
    // __tests__/__snapshots__

    // the minute ArticleList is changed, snapshot fails
    expect(tree).toMatchSnapshot();
  });
});
