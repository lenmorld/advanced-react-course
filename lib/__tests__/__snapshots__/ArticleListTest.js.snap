// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderColor": "gray",
        "borderStyle": "solid",
        "borderWidth": "1px",
        "padding": "21px",
      }
    }
  >
    <div
      style={Object {}}
    />
    <div
      style={Object {}}
    >
      Invalid Date
    </div>
    <div
      style={Object {}}
    >
      <a>
         
      </a>
    </div>
    <div
      style={Object {}}
    />
  </div>
  <div
    style={
      Object {
        "borderColor": "gray",
        "borderStyle": "solid",
        "borderWidth": "1px",
        "padding": "21px",
      }
    }
  >
    <div
      style={Object {}}
    />
    <div
      style={Object {}}
    >
      Invalid Date
    </div>
    <div
      style={Object {}}
    >
      <a>
         
      </a>
    </div>
    <div
      style={Object {}}
    />
  </div>
</div>
`;
